include/rmm/mr/device/arena_memory_resource.hpp

```diff
@@ -335,7 +335,7 @@ class arena_memory_resource final : public device_memory_resource {
   void dump_memory_log(size_t bytes)
   {
     logger_->info("**************************************************");
-    logger_->info("Ran out of memory trying to allocate {}.", rmm::detail::bytes{bytes});
+    logger_->info("Ran out of memory trying to allocate %s.", rmm::detail::bytes{bytes});
```

Comment:
Have you tested that this works? In my testing so far, this puts a literal `%s` into the log file.

Reply:
I verified that printf format placeholders are not supported in spdlog (or fmt). https://github.com/gabime/spdlog/issues/1843

Reply:
This call doesn't go straight to spdlog. It gets piped to the logger::log implementation in our new logger header, which calls std::snprintf and then passes a fully formatted string to spdlog's log API (with no fmt-style or C-style format strings).

Reply:
Sorry, missed that.

Reply:
Oh I see, I somehow lost the conditional compilation here. My apologies. I'll fix that.